### PR TITLE
Several followup fixes from #1660

### PR DESCRIPTION
**Pull request by olafurpg**
See commit messages for details.

### CODE REVIEW

**Comment by olafurpg** [comment]
I added the `all` scope because while debugging `ConvertSuite` I was frustrated that `-- -z TEST_NAME` doesn't work with slow tests. I then updated CONTRIBUTING.md to document this change for for other people.

File: semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala [hunk 1]
```diff
@@ -15,10 +15,12 @@ import scala.meta.metacp._
 import scala.util.control.NonFatal
 import java.util.concurrent.atomic.AtomicBoolean
 import scala.collection.GenSeq
+import scala.collection.mutable
 
 class Main(settings: Settings, reporter: Reporter) {
 
   lazy val classpathIndex = ClasspathIndex(settings.fullClasspath)
+  private val isReportedMissingSymbol = mutable.Set.empty[String]
```
Comment (xeno-by) [inline semanticdb/metacp/src/main/scala/scala/meta/internal/metacp/Main.scala]: How about renaming this to `missingSymbols`?

File: semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/SymbolInformationOps.scala [hunk 2]
```diff
@@ -53,23 +53,14 @@ trait SymbolInformationOps { self: Scalacp =>
...
-          if (sym.isPackageAccordingToClasspath) k.PACKAGE
-          else if (hasTermName || isScalaObject) k.OBJECT
-          else k.CLASS
+          if (isModuleClass) {
+            if (sym.isPackageAccordingToClasspath) k.PACKAGE
+            else if (sym.isJavaDefined) k.CLASS
+            else k.OBJECT
+          } else {
+            k.CLASS
+          }
```
Comment (xeno-by) [inline semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/SymbolInformationOps.scala]: 😍

File: semanticdb/metacp/src/main/scala/scala/meta/metacp/Settings.scala [hunk 3]
```diff
@@ -11,9 +11,7 @@ final class Settings private (
     val classpath: Classpath,
     val dependencyClasspath: Classpath,
     val scalaLibrarySynthetics: Boolean,
-    val par: Boolean,
-    val assumeJava: Set[String],
-    val assumeScala: Set[String]
```
Comment (xeno-by) [inline semanticdb/metacp/src/main/scala/scala/meta/metacp/Settings.scala]: 😍

File: tests/jvm/src/test/resources/metac-metacp.diff [hunk 4]
```diff
@@ -309,24 +309,13 @@ com/javacp/ClassSuffix#suffix.
 -                    symbol: "com/javacp/ClassSuffix#"
                  symbol: "com/javacp/ClassSuffix#Inner#"
                  type_arguments {
--            symbol: "com/javacp/ClassSuffix#Inner#Bar#"
-+            symbol: "_empty_/Bar#"
--        symbol: "com/javacp/ClassSuffix#Inner#Bar#Fuz#"
-+        symbol: "_empty_/Fuz#"
-         type_arguments {
-           typeRef {
-             prefix {
```
Comment (xeno-by) [inline tests/jvm/src/test/resources/metac-metacp.diff]: This seems to fix https://github.com/scalameta/scalameta/issues/1554?

File: tests/jvm/src/test/resources/metac-metacp.diff [hunk 5]
```diff
@@ -309,24 +309,13 @@ com/javacp/ClassSuffix#suffix.
...
 
 
 ================
 com/javacp/Coin#
 ================
 --- metac
 +++ metacp
-     self {
-     declarations {
-+      symlinks: "com/javacp/Coin#$VALUES."
```
Comment (xeno-by) [inline tests/jvm/src/test/resources/metac-metacp.diff]: 😍

File: tests/jvm/src/test/scala/scala/meta/tests/metacp/MetacpOps.scala [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          visitType(upperBound)
+        }
+      case s.ValueSignature(tpe) =>
+        visitType(tpe)
+      case s.NoSignature =>
+        ()
+    }
+    visitSignature(info.signature)
+    references.result()
+  }
```
Comment (xeno-by) [inline tests/jvm/src/test/scala/scala/meta/tests/metacp/MetacpOps.scala]: The new type API is awesome :)

**Pull request merged**